### PR TITLE
Avoid infinite recursion when renaming #3326

### PR DESCRIPTION
**Pull request by rafatower**
See the ticket for more info.

@Kartones please review

### CODE REVIEW

File: app/models/table.rb [hunk 1]
```diff
@@ -1337,27 +1337,24 @@ def get_table_id
...
         end
       end
 
-      propagate_namechange_to_table_vis unless errored
+      propagate_namechange_to_table_vis
+
 
-      unless errored
         if @user_table.layers.blank?
           exception_to_raise = CartoDB::TableError.new("Attempt to rename table without layers #{qualified_table_name}")
```
Comment (Kartones) [inline app/models/table.rb]: I'd either change to raise error (as we're not launching/raising an exception), or as you told me in person, directly remove this par of the if and let the code explode and/or be treated at upper layers if layers are empty.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3220/))

**Comment by Kartones** [comment]
:+1:  Apart from that `exception_to_raise` only used for notification and not actual raising

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3223/))

**Comment by rafatower** [comment]
I tested 2e1d70e but in the end I prefer to keep the `CartoDB::notify_exception` so that it appends the user info to the trace. In case of weirdness it can greatly help us reconstruct what happened.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3225/))

**Pull request merged**